### PR TITLE
[bug](function) fix milliseconds_diff function return wrong result

### PR DESCRIPTION
**Pull request by zhangstar333**
## Proposed changes

the microsecond_diff function is used by:
INT microseconds_diff(DATETIME enddate, DATETIME startdate)
TIME TIMEDIFF(DATETIME expr1, DATETIME expr2)
as the time type value between '-838:59:59' and '838:59:59', so the return value should limited,
but when return it's int, shouldn't have any limit.

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zhangstar333** [comment]
run buildall

File: be/src/vec/functions/function_date_or_datetime_computation.h [hunk 1]
```diff
@@ -295,7 +295,13 @@ struct TimeDiffImpl {
         const auto& ts1 = reinterpret_cast<const DateValueType2&>(t1);
         is_null = !ts0.is_valid_date() || !ts1.is_valid_date();
         if constexpr (UsingTimev2) {
-            return ts0.microsecond_diff(ts1);
+            // refer to https://dev.mysql.com/doc/refman/5.7/en/time.html
+            // the time type value between '-838:59:59' and '838:59:59', so the return value should limited
+            int64_t diff_m = ts0.microsecond_diff(ts1);
+            if (diff_m > (int64_t)3020399 * 1000 * 1000) {
```
Comment (zclllyybb) [inline be/src/vec/functions/function_date_or_datetime_computation.h:301]: is there a need to check the lower_bound?

**Comment by zhangstar333** [comment]
run buildall

File: be/src/vec/functions/function_date_or_datetime_computation.h [hunk 2]
```diff
@@ -289,15 +289,25 @@ struct TimeDiffImpl {
 
     static constexpr auto name = "timediff";
     static constexpr auto is_nullable = false;
+    static constexpr int64_t limit_value = 3020399000000;
```
Comment (HappenLee) [inline be/src/vec/functions/function_date_or_datetime_computation.h:292]: Add comment for the value

**Comment by zhangstar333** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by Mryange — APPROVED** [review]

**Pull request merged**